src/components/legend/color.tsx

```diff
@@ -34,32 +33,34 @@ interface ColorProps {
  * @internal
  */
 export const Color = memo(
-  forwardRef<HTMLDivElement, ColorProps>(({ color, isSeriesHidden = false, hasColorPicker, onClick }, ref) => {
+  forwardRef<HTMLButtonElement, ColorProps>(({ color, isSeriesHidden = false, hasColorPicker, onClick }, ref) => {
     if (isSeriesHidden) {
       return (
-        <button type="button" className="echLegendItem__color" aria-label="series hidden" title="series hidden">
+        <div className="echLegendItem__color" title="series hidden">
           {/* changing the default viewBox for the eyeClosed icon to keep the same dimensions */}
-          <Icon type="eyeClosed" viewBox="-3 -3 22 22" />
-        </button>
+          <Icon type="eyeClosed" viewBox="-3 -3 22 22" aria-label="series color is hidden" />
+        </div>
       );
     }
 
-    const colorClasses = classNames('echLegendItem__color', {
-      'echLegendItem__color--changable': hasColorPicker,
-    });
+    if (hasColorPicker) {
+      return (
+        <button
+          type="button"
+          onClick={onClick}
+          className="echLegendItem__color--changable"
+          title="change series color"
+          ref={ref}
+        >
+          <Icon type="dot" color={color} aria-label="change series color" />
+        </button>
+      );
+    }
 
     return (
-      <button
-        type="button"
-        onClick={hasColorPicker ? onClick : undefined}
-        className={colorClasses}
-        aria-label={hasColorPicker ? 'change series color' : 'series color'}
-        title={hasColorPicker ? 'change series color' : 'series color'}
-      >
-        <div ref={ref}>
-          <Icon type="dot" color={color} />
-        </div>
-      </button>
+      <div className="echLegendItem__color" title="series color">
+        <Icon type="dot" color={color} aria-label={hasColorPicker ? 'change series color' : 'series color'} />
```

Comment:
Can remove the `aria-label` conditional, and would be good to add a color name here

```suggestion
        <Icon type="dot" color={color} aria-label={`series color: ${colorName}`} />
```

Reply:
Please see bc14508